lib.php

```diff
@@ -144,6 +154,16 @@ function mass_enroll($cir, $course, $context, $data) {
 
         // Iterate over each user account found.
         foreach ($users as $user) {
+            if ($user->deleted) {
+                $result .= get_string('im:user_deleted', 'local_mass_enroll', fullname($user)) . "\n";
+                continue;
+            }
+
+            if ($user->username == 'guest') {
```

Comment:
I think we should also check against the real guest object. Matching on a constant here feels brittle.

```
$guest = guest_user();
if ($user->username === $guest->username) {
  continue;
}
```